File: flutter_app/lib/features/model/agent_context_info.dart
class AgentContextInfo {
  final String? operatorLabel;
  final String? accessScopeLabel;
  final bool authenticated;
  final bool privilegedOperator;

  const AgentContextInfo({
    this.operatorLabel,
    this.accessScopeLabel,
    this.authenticated = false,
    this.privilegedOperator = false,
  });

  factory AgentContextInfo.fromJson(Map<String, dynamic> json) {
    return AgentContextInfo(
      operatorLabel: json['operatorLabel'] as String?,
      accessScopeLabel: json['accessScopeLabel'] as String?,
      authenticated: json['authenticated'] == true,
      privilegedOperator: json['privilegedOperator'] == true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'operatorLabel': operatorLabel,
      'accessScopeLabel': accessScopeLabel,
      'authenticated': authenticated,
      'privilegedOperator': privilegedOperator,
    };
  }
}

File: backend/src/main/java/com/tutict/finalassignmentbackend/model/ai/AgentContextInfo.java
package com.tutict.finalassignmentbackend.model.ai;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.tutict.finalassignmentbackend.service.agent.AgentSkillContext;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record AgentContextInfo(
        String operatorLabel,
        String accessScopeLabel,
        boolean authenticated,
        boolean privilegedOperator
) {

    public static AgentContextInfo from(AgentSkillContext context) {
        return new AgentContextInfo(
                context.operatorLabel(),
                context.accessScopeLabel(),
                context.isAuthenticated(),
                context.isPrivilegedOperator()
        );
    }
}
